### PR TITLE
Fix pound i

### PR DESCRIPTION
**Pull request by KevinRansom**
A better fix and couple more tests.

### CODE REVIEW

File: src/Microsoft.DotNet.Interactive.Tests/LanguageKernelPackageTests.cs [hunk 1]
```diff
@@ -425,18 +425,68 @@ public async Task Pound_i_nuget_displays_list_of_added_sources(Language language
...
-                  .ContainSingle<DisplayedValueUpdated>()
-                  .Which
-                  .FormattedValues
-                  .Should()
-                  .ContainSingle(v => v.MimeType == "text/html")
-                  .Which
-                  .Value
-                  .Should()
-                  .ContainAll("Restore sources", "https://completelyFakerestoreSource");
+            Assert.Collection(events,
```
Comment (jonsequitur) [inline src/Microsoft.DotNet.Interactive.Tests/LanguageKernelPackageTests.cs:433]: Could you use FluentAssertions and make the specification a little more focused? Maybe try `events.Should().ContainInOrder(...)` and only specify the events we care about?

File: src/Microsoft.DotNet.Interactive.Tests/LanguageKernelPackageTests.cs [hunk 2]
```diff
@@ -425,18 +425,68 @@ public async Task Pound_i_nuget_displays_list_of_added_sources(Language language
...
+
+            using var events = kernel.KernelEvents.ToSubscribedList();
+
+            await kernel.SubmitCodeAsync(@"
+#i ""nuget:https://completelyFakerestoreSourceCommand1.1""
+#i ""nuget:https://completelyFakerestoreSourceCommand1.2""
+");
+
+            var displayValueProducedEvents = KernelEvents.Where(e => e is DisplayedValueProduced);
+            Assert.True(displayValueProducedEvents.Count() == 1);
```
Comment (jonsequitur) [inline src/Microsoft.DotNet.Interactive.Tests/LanguageKernelPackageTests.cs:462]: Please remove the intermediate sanity check assertions.

File: src/Microsoft.DotNet.Interactive.Tests/LanguageKernelPackageTests.cs [hunk 3]
```diff
@@ -425,18 +425,68 @@ public async Task Pound_i_nuget_displays_list_of_added_sources(Language language
...
+            var displayValueUpdatedEvents = KernelEvents.Where(e => e is DisplayedValueUpdated);
+            Assert.True(displayValueProducedEvents.Count() == 1);
+
+            await kernel.SubmitCodeAsync(@"
+#i ""nuget:https://completelyFakerestoreSourceCommand2.1""
+#i ""nuget:https://completelyFakerestoreSourceCommand2.2""
+");
+
+            // Only produces a single DisplayedValueProduced event
+            displayValueProducedEvents = KernelEvents.Where(e => e is DisplayedValueProduced);
```
Comment (jonsequitur) [inline src/Microsoft.DotNet.Interactive.Tests/LanguageKernelPackageTests.cs:473]: This test doesn't look like it's testing something different from the one above.

File: src/Microsoft.DotNet.Interactive.Tests/LanguageKernelPackageTests.cs (hunk 1 above)
Comment (KevinRansom) [reply src/Microsoft.DotNet.Interactive.Tests/LanguageKernelPackageTests.cs:433]: @jonsequitur , lordy ... I am not fluent in fluentassertions, but yes, I can try.

File: src/Microsoft.DotNet.Interactive.Tests/LanguageKernelPackageTests.cs (hunk 3 above)
Comment (KevinRansom) [reply src/Microsoft.DotNet.Interactive.Tests/LanguageKernelPackageTests.cs:473]: It verifies that 2 submissions accumulate the output.

Comment (jonsequitur) [reply src/Microsoft.DotNet.Interactive.Tests/LanguageKernelPackageTests.cs:473]: Ah, I see. I missed that.

Comment (jonsequitur) [reply src/Microsoft.DotNet.Interactive.Tests/LanguageKernelPackageTests.cs:473]: Can we make the test name more explicit in that case?

**Review by jonsequitur — APPROVED** [review]

File: src/Microsoft.DotNet.Interactive.Tests/LanguageKernelPackageTests.cs (hunk 3 above)
Comment (KevinRansom) [reply src/Microsoft.DotNet.Interactive.Tests/LanguageKernelPackageTests.cs]: Yes ... naming and cache eviction the two hardest problems in computer science.  Not that what I do is science. :-)

**Pull request merged**